web/gui/dashboard/src/domains/chart/sagas.ts

```diff
@@ -0,0 +1,45 @@
+import { takeEvery, put, call } from "redux-saga/effects"
+
+import { axiosInstance } from "utils/api"
+
+import { fetchDataAction, FetchDataPayload } from "./actions"
+
+type FetchDataSaga = { payload: FetchDataPayload }
+function* fetchDataSaga({ payload }: FetchDataSaga) {
+  const {
+    // props for api
+    chart, format, points, group, gtime, options, after, before,
+    // props for the store
+    fetchDataParams, id,
+  } = payload
+  let response
+  try {
```

Comment:
Maybe the try-catch thing could be moved to separate module, just not to distract from reading effects handling, as it's an independent implementation detail?